consumer.go

```diff
@@ -158,33 +158,26 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 	defer scanTicker.Stop()
 
 	for {
-		select {
-		case <-ctx.Done():
-			return nil
-		case <-scanTicker.C:
-			resp, err := c.client.GetRecords(&kinesis.GetRecordsInput{
-				Limit:         aws.Int64(c.maxRecords),
-				ShardIterator: shardIterator,
-			})
+		resp, err := c.client.GetRecords(&kinesis.GetRecordsInput{
+			Limit:         aws.Int64(c.maxRecords),
+			ShardIterator: shardIterator,
+		})
 
-			// attempt to recover from GetRecords error when expired iterator
-			if err != nil {
-				c.logger.Log("[CONSUMER] get records error:", err.Error())
-
-				if awserr, ok := err.(awserr.Error); ok {
-					if _, ok := retriableErrors[awserr.Code()]; !ok {
-						return fmt.Errorf("get records error: %v", awserr.Message())
-					}
-				}
+		// attempt to recover from GetRecords error when expired iterator
+		if err != nil {
+			c.logger.Log("[CONSUMER] get records error:", err.Error())
 
-				shardIterator, err = c.getShardIterator(ctx, c.streamName, shardID, lastSeqNum)
-				if err != nil {
-					return fmt.Errorf("get shard iterator error: %v", err)
+			if awserr, ok := err.(awserr.Error); ok {
+				if _, ok := retriableErrors[awserr.Code()]; !ok {
+					return fmt.Errorf("get records error: %v", awserr.Message())
 				}
-
-				continue
 			}
 
+			shardIterator, err = c.getShardIterator(ctx, c.streamName, shardID, lastSeqNum)
+			if err != nil {
+				return fmt.Errorf("get shard iterator error: %v", err)
+			}
+		} else {
```

Comment:
this might be out of the scope of this PR, but thoughts on removing this `else` to early continue 

```go

if err != nil {
  // ...

  shardIterator, err = c.getShardIterator(ctx, c.streamName, shardID, lastSeqNum)
  if err != nil {
    return fmt.Errorf("get shard iterator error: %v", err)
  }
  continue
}

// loop over records, call callback func
for _, r := range resp.Records {
  // ...
}

// Wait for next scan
select {
case <-ctx.Done():
  return nil
case <-scanTicker.C:
  continue
}
```

Reply:
Would definitely make the code cleaner.  There's little downside, except that we would be running the extra call to the AWS API even if we've been asked to shut down.